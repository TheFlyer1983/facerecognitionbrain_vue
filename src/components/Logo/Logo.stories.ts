import { Story } from '@storybook/vue3';

import Logo from './Logo.vue';

export default {
  title: 'Logo',
  component: Logo
};

const Template = () => ({
  components: { Logo },
  template: `<Logo />`
});

export const Default: Story = Template.bind({});
