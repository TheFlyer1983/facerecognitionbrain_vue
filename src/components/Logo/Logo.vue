<script setup lang="ts">
import brain from '@/assets/images/brain.png';
</script>

<template>
  <div class="logo">
    <div class="logo-tilt">
      <div class="logo-tilt-inner">
        <img :src="brain" alt="logo" class="logo-tilt-inner--image" />
      </div>
    </div>
  </div>
</template>

<style scoped lang="scss">
.logo {
  text-align: center;
  margin: 2rem;
  margin-top: 0;

  &-tilt {
    background: linear-gradient(89deg, #ff5edf 0%, #04c8de 100%);
    border-radius: 0.25rem;
    box-shadow: 0 0 8px 2px rgba(0, 0, 0, 0.2);
    height: 150px;
    width: 150px;

    &-inner {
      padding: 1rem;

      &--image {
        padding-top: 5px;
      }
    }
  }
}
</style>
