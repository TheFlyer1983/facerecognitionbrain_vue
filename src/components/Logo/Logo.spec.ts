import { beforeEach, describe, expect, it } from 'vitest';
import { shallowMount, VueWrapper } from '@vue/test-utils';
import Logo from './Logo.vue';

describe('Given the Logo component', () => {
  const render = () => shallowMount(Logo);
  let wrapper: VueWrapper<InstanceType<typeof Logo>>;

  describe('when it is rendered', () => {
    beforeEach(() => {
      wrapper = render();
    });
    
    it('should match the snapshot', () => {
      expect(wrapper.element).toMatchSnapshot();
    });
  });
});
