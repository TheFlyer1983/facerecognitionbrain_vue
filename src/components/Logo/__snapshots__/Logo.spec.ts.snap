// Vitest Snapshot v1

exports[`Given the Logo component > when it is rendered > should match the snapshot 1`] = `
<div
  class="logo"
  data-v-4bfea8c3=""
>
  <div
    class="logo-tilt"
    data-v-4bfea8c3=""
  >
    <div
      class="logo-tilt-inner"
      data-v-4bfea8c3=""
    >
      <img
        alt="logo"
        class="logo-tilt-inner--image"
        data-v-4bfea8c3=""
        src="/src/assets/images/brain.png"
      />
    </div>
  </div>
</div>
`;
